src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -285,7 +287,9 @@ internal static bool ReportConflictWithParameter(Symbol parameter, Symbol newSym
 
             if (parameterKind == SymbolKind.TypeParameter)
             {
-                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local)
+                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local ||
+                    (newSymbolKind == SymbolKind.Method && 
+                     ((MethodSymbol)newSymbol).MethodKind == MethodKind.LocalFunction))
                 {
                     // CS0412: 'X': a parameter or local variable cannot have the same name as a method type parameter
                     diagnostics.Add(ErrorCode.ERR_LocalSameNameAsTypeParam, newLocation, name);
```

Comment:
Consider updating the comment in the use of `ERR_LocalDuplicate` in `LocalScopeBinder` as well.